services/github/github-total-star.service.js

```diff
@@ -0,0 +1,259 @@
+      }).required(),
+    })
+  )
+  .default([])
+
+const repositoriesSchema = Joi.object({
+  pageInfo: pageInfoSchema,
+  nodes: nodesSchema,
+}).required()
+
+const schema = Joi.object({
+  data: Joi.alternatives(
+    Joi.object({
+      user: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required(),
+    Joi.object({
+      organization: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required()
+  ).required(),
+}).required()
+
+const userQuery = gql`
+  query fetchStars(
+    $user: String!
+    $nextCursor: String
+    $affiliations: [RepositoryAffiliation]!
+  ) {
+    user(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        affiliations: $affiliations
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const orgQuery = gql`
+  query fetchStars($user: String!, $nextCursor: String) {
+    organization(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+const affiliationsAllowedValues = [
+  'OWNER',
+  `COLLABORATOR`,
+  'ORGANIZATION_MEMBER',
+]
+/**
+ * Validates affiliations should contain combination of allowed values in any order.
+ *
+ * @param {string} value affiliation current value
+ * @param {*} helpers object to construct custom error
+ *
+ * @returns {string} valiadtion error or value unchanged
+ */
+const validateAffiliations = (value, helpers) => {
+  const values = value.split(',')
+  if (values.some(e => !affiliationsAllowedValues.includes(e))) {
+    return helpers.error('any.invalid')
+  }
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  org: Joi.valid('').optional(),
+  // TODO add regex
+  affiliations: Joi.string().default('OWNER').custom(validateAffiliations),
+}).required()
+
+module.exports = class TotalStarService extends GithubAuthV4Service {
+  static get defaultLabel() {
+    return 'Total Stars'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'github/users',
+      pattern: ':user/total-stars',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: "Github User's total stars",
+        namedParams: {
+          user: 'chris48s',
+        },
+        queryParams: { affiliations: 'OWNER,COLLABORATOR' },
+        staticPreview: {
+          label: this.defaultLabel,
+          message: 54,
+          style: 'social',
+        },
+        documentation: userDocumentation,
```

Comment:
This example doesn't work for me on the review app, nor do any other users i've tried

https://shields-staging-pr-5507.herokuapp.com/github/users/chris48s/total-stars?style=social
![](https://shields-staging-pr-5507.herokuapp.com/github/users/chris48s/total-stars?style=social)

https://shields-staging-pr-5507.herokuapp.com/github/users/chris48s/total-stars?affiliations=OWNER%2CCOLLABORATOR&style=social
![](https://shields-staging-pr-5507.herokuapp.com/github/users/chris48s/total-stars?affiliations=OWNER%2CCOLLABORATOR&style=social)

https://shields-staging-pr-5507.herokuapp.com/github/users/sindresorhus/total-stars?affiliations=OWNER%2CCOLLABORATOR&style=social
![](https://shields-staging-pr-5507.herokuapp.com/github/users/sindresorhus/total-stars?affiliations=OWNER%2CCOLLABORATOR&style=social)

Reply:
It should work now. A little confusing fields `affiliations` and `ownerAffiliations` were there in graphql. Github doc didn't explained which one to use nd when. When I used `ownerAffiliations`, it worked for my account as well as other account.
`affiliations` allowed filtering on my account only, and for other accounts, without throwing any errors, it retuerned zero repos.